src/libbson/src/bson/bson-string.c

```diff
@@ -126,17 +134,22 @@ bson_string_append (bson_string_t *string, /* IN */
                     const char *str)       /* IN */
 {
    uint32_t len;
+   size_t len_sz;
 
    BSON_ASSERT (string);
    BSON_ASSERT (str);
 
-   len = (uint32_t) strlen (str);
+   len_sz = strlen (str);
+   BSON_ASSERT (bson_in_range_unsigned (uint32_t, len_sz));
+   len = (uint32_t) len_sz;
 
    if ((string->alloc - string->len - 1) < len) {
+      BSON_ASSERT (bson_in_range_unsigned (uint32_t, string->alloc + len));
```

Comment:
Similar to above, `string->alloc + len` may overflow. Suggested alternative:

```suggestion
      BSON_ASSERT (string->alloc <= UINT32_MAX - len);
```